library/alloc/src/vec/spec_extend.rs

```diff
@@ -47,7 +47,12 @@ where
                 });
```

Comment:
Why does the above code seemed to have the same condition written twice? See 10 lines above.

Reply:
One checks that high == low. the other checks that it's finite.

Reply:
No, I mean
```rust
        if let Some(high_value) = high {
            debug_assert_eq!(
                low,
                high_value,
                "TrustedLen iterator's size hint is not exact: {:?}",
                (low, high)
            );
        }
        if let Some(additional) = high {
            self.reserve(additional);
...
```

Reply:
idk, probably just an historic artifact of code being moved around and redundant conditions not being merged. Someone just has to clean it up.